.github/workflows/auto_cherry_pick.yml

```diff
@@ -7,37 +7,67 @@ on:
-  auto_cherry_picking:
+      - id: set-branch
+        run: echo "::set-output name=previous_branch::$(if [ $GITHUB_BASE_REF  == 'master' ]; then echo $(git branch -rl 'origin/6.*.z' --format='%(refname:lstrip=-1)' | sort --version-sort | tail -n1 | jq -cnR '[inputs | select(length>0)]'); else echo ['"6.'$(($(echo $GITHUB_BASE_REF | cut -d. -f2) - 1))'.z"']; fi)"
+
+  auto-cherry-pick:
+    name: Auto Cherry Pick to previous branch
     if: github.event.pull_request.merged == true && contains(github.event.pull_request.labels.*.name, 'CherryPick')
-    name: Auto Cherry picking
-    needs: branch-matrix
+    needs: previous-branch
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        to_branch: ${{ fromJson(needs.branch-matrix.outputs.branches) }}
+        to_branch: ${{ fromJson(needs.previous-branch.outputs.previous_branch) }}
     steps:
-      - name: Checkout Nailgun
-        uses: actions/checkout@v3
+      - uses: actions/checkout@v3
         with:
           fetch-depth: 0
-        if: matrix.to_branch != github.base_ref
       - name: Cherry pick into ${{ matrix.to_branch }}
-        uses: carloscastrojumo/github-cherry-pick-action@v1.0.2
+        uses: jyejare/github-cherry-pick-action@main
         with:
+          token: ${{ secrets.CHERRYPICK_PAT }}
           branch: ${{ matrix.to_branch }}
           labels: |
             Auto_Cherry_Picked
-        # skipping PRs remote target_branch from cherrypicking into itself
-        if: matrix.to_branch != github.base_ref
+            ${{ matrix.to_branch }}
+          assignees: |
+            ${{ github.event.pull_request.user.login }}
+
+  create-issue:
+    runs-on: ubuntu-latest
+    if: ${{ always() && contains(join(needs.*.result, ','), 'failure') }}
+    needs: auto-cherry-pick
+    steps:
+      - name: Create Issue on Failed Auto Cherrypick
+        uses: dacbd/create-issue-action@main
+        with:
+          token: ${{ secrets.CHERRYPICK_PAT }}
+          title: "[Failed-AutoCherryPick] - ${{ github.event.pull_request.title }}"
+          body: |
+            #### Auto-Cherry-Pick WorkFlow Failure:
+            - [Failed Cherrypick Action](https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }})
+            - [Parent Pull Request](https://github.com/${{ github.repository }}/pull/${{ github.event.number }})
+          labels: Failed_AutoCherryPick
+          assignees: ${{ github.event.pull_request.user.login }}
+
+  google-chat-notification:
+    runs-on: ubuntu-latest
+    if: ${{ always() && contains(join(needs.*.result, ','), 'failure') }}
+    needs: auto-cherry-pick
+    steps:
+      - name: Google Chat Notification - Nailgun
+        uses: Co-qn/google-chat-notification@releases/v1
+        with:
+          name: "[Failed-AutoCherryPick] - ${{ github.event.pull_request.title }}"
```

Comment:
I think it would be better to have some diff here in the naming of card notification
```suggestion
          name: "[${{ github.repository }}-AutoCherryPick-Failed] - ${{ github.event.pull_request.title }}"
```

Reply:
Not required the notification shows the repository :

<img width="515" alt="Screenshot 2022-11-03 at 12 43 44 PM" src="https://user-images.githubusercontent.com/11752425/199665164-afa79d22-2b50-4686-b80a-04efc42305f5.png">